### PR TITLE
add datastax connection metrics to our existing metrics registry

### PR DESCRIPTION
**Pull request by shintasmith**
**WHAT**
Datastax connection has its own collection of metrics, as documented here:
http://docs.datastax.com/en/drivers/java/3.0/com/datastax/driver/core/Metrics.html

This PR is to grab all of those metrics and register them with our Blueflood MetricsRegistry, so we have metrics related to datastax in our graphite. 

I started Blueflood with Datastax configured. Then from JConsole, I was able to see all the datastax.* metrics under our metrics.

### CODE REVIEW

**Comment by shintasmith** [comment]
@usnavi @ChandraAddala @VinnyQ @izrik pls review

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxIO.java [hunk 1]
```diff
@@ -113,34 +106,58 @@ private static PoolingOptions getPoolingOptions(int numHosts){
...
+            public void onTimerAdded(String name, Timer timer) {
+                bfMetricsRegistry.register(metricsPrefix + name, timer);
+            }
+
+            @Override
+            public void onTimerRemoved(String name) {
+                bfMetricsRegistry.remove(metricsPrefix + name);
             }
-        }, 1, 1, TimeUnit.MINUTES);
+        });
```
Comment (ChandraAddala) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxIO.java]: I am having hard time understand the relation between code being removed and the new code added. Can you explain it a little bit?

**Comment by ChandraAddala** [comment]
Looks good.

**Comment by usnavi** [comment]
lgtm

**Comment by izrik** [comment]
@shintasmith Would we want to specify separate prefixes for ingest vs. query vs. rollup? They all use datastax, right?

**Comment by shintasmith** [comment]
@izrik, good point, but I am not sure we separate our metrics that way today with Astyanax. All of them just go to the same bucket.

**Comment by izrik** [comment]
@shintasmith Fair enough. I just wanted to bring that up, in case it might become an issue later. Other than that, it looks good to me.

**Pull request merged**